connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -324,13 +325,11 @@ private PluginScanResult scanPluginPath(
         return result;
     }
 
-    private <T> Collection<PluginDesc<T>> getServiceLoaderPluginDesc(Class<T> klass,
-                                                                     ClassLoader loader) {
-
+    private <T> Collection<PluginDesc<T>> getServiceLoaderPluginDesc(Class<T> klass, ClassLoader loader) {
         ServiceLoader<T> serviceLoader = ServiceLoader.load(klass, loader);
         Collection<PluginDesc<T>> result = new ArrayList<>();
         for (T impl : serviceLoader) {
-            result.add(new PluginDesc<>(klass, versionFor(impl), loader));
+            result.add(new PluginDesc<>((Class<? extends T>) impl.getClass(), versionFor(impl), loader));
```

Comment:
Since we are changing lines in this loop, I'd like to suggest a renaming: `impl` -> `serviceType`. 
`impl` seems too vague and also a bit inaccurate since it might correspond to an interface (which usually is not referred to as `implementation`).

Reply:
@kkonstantine It is actually an implementation that we get back from the service loader. But I do agree, it is way too generic. Perhaps pluginImpl?